packages/jaeger-ui-components/src/TraceTimelineViewer/SpanBarRow.test.js

```diff
@@ -139,21 +133,18 @@ describe('<SpanBarRow>', () => {
       },
       props.span
     );
-    const spanRow = shallow(
+    render(
       <SpanBarRow
         {...props}
         span={span}
         createSpanLink={() => ({
           traceLinks: [{ content: 'This span is referenced by another span', href: 'href' }],
         })}
       />
-    )
-      .dive()
-      .dive()
-      .dive();
-    const menu = spanRow.find(`a[href="href"]`);
-    expect(menu.length).toEqual(1);
-    expect(menu.at(0).text()).toEqual('This span is referenced by another span');
+    );
+    expect(screen.getAllByTestId('a-href')).toHaveLength(1);
+    const { getByText } = within(screen.getByTestId('a-href'));
+    expect(getByText('This span is referenced by another span')).toBeInTheDocument();
```

Comment:
we don't need to add a testid here

```suggestion
    expect(screen.getByRole('link', { name: 'This span is referenced by another span' })).toBeInTheDocument();
```